### PR TITLE
[xrt] Add a simple cc_binary server

### PR DESCRIPTION
**Pull request by Keno**
Right now xrt isn't included in any of the standard ways to
spin up a grpc server, making it tough to play with it without
making any local modifications to the build system. This patch
adds a small server binary that includes xrt and simply listens
on `localhost:8470`.

### CODE REVIEW

**Comment by qlzh727** [comment]
Michael, can u take look? Thanks.

**Comment by mrry** [comment]
@michaelisard This one maybe?

https://github.com/tensorflow/tensorflow/blob/e1a32c98210f8ebba42a0397259d948e1433c09e/tensorflow/core/distributed_runtime/rpc/BUILD#L305-L320

The most common way to start a server in OSS is by creating a `tf.train.Server` from Python. Currently it only configures a `MasterService` and `WorkerService`, but it wouldn't be too difficult to allow it to start other services as well/instead.

**Comment by michaelisard** [comment]
@mrry do you think it would be reasonable to put grpc_tensorflow_server.cc into a library that Keno could put into his build rule, so that he could avoid adding any C++ here (i.e., stop duplicating the code for main() here)? Or is it better just to duplicate it and leave things alone in the rpc directory?

**Comment by michaelisard** [comment]
Actually I should also check: @keno if you deleted xrt_server.cc and linked against grpc_tensorflow_server.cc in your build rule instead, would your use case work?

**Comment by mrry** [comment]
I think that should work, but I don't mind either way. One caveat, the command-line args to grpc_tensorflow_server.cc weren't designed for ease-of-use... just for testing. In particular, there's no simple flag for "just run it locally and print out what port you chose" or "just run it locally on a particular port" options. It might be desirable to add those :).

**Comment by michaelisard** [comment]
OK, @Keno I would marginally prefer not to add xrt_server.cc because if it gets checked in then it will have to be supported forever, but I don't feel strongly enough to block you making progress if it's too inconvenient for you to switch to using grpc_tensorflow_server.cc. Let me know what you think!

**Comment by Keno** [comment]
Sorry for the delay here. I'm happy to use `grpc_tensorflow_server.cc` (and maybe add convenience extras to those as necessary). However, I'm not sure my bazel-foo is sufficient to write those build rules correctly (to have a build option to build that .cc with xrt).

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
